src/Components/DatePicker/Base/DatePicker.tsx

```diff
@@ -0,0 +1,183 @@
+// Libraries
+import React, {PureComponent, ChangeEvent} from 'react'
+import ReactDatePicker from 'react-datepicker'
+import moment from 'moment'
+
+// Components
+import {Input} from '../../Inputs/Input'
+import {Grid} from '../../GridLayout/Grid'
+import {Form} from '../../FormLayout/Form'
+
+// Styles
+import '../DatePicker.scss'
+
+// Types
+import {
+  Columns,
+  ComponentSize,
+  ComponentStatus,
+  StandardProps,
+} from '../../../Types'
+
+interface Props extends StandardProps {
+  /** Label for input field */
+  label: string
+  /** Date value */
+  dateTime?: string | null
+  /** Function called once a date is selected */
+  onSelectDate: (date: string) => void
+}
+
+interface State {
+  inputValue: string
+  inputFormat: string
+}
+
+const isValidRTC3339 = (d: string): boolean => {
+  return (
+    moment(d, 'YYYY-MM-DD HH:mm', true).isValid() ||
+    moment(d, 'YYYY-MM-DD HH:mm:ss', true).isValid() ||
+    moment(d, 'YYYY-MM-DD HH:mm:ss.SSS', true).isValid() ||
+    moment(d, 'YYYY-MM-DD', true).isValid()
+  )
+}
+
+const getFormat = (d: string): string | undefined => {
+  if (moment(d, 'YYYY-MM-DD', true).isValid()) {
+    return 'YYYY-MM-DD'
+  }
+  if (moment(d, 'YYYY-MM-DD HH:mm', true).isValid()) {
+    return 'YYYY-MM-DD HH:mm'
+  }
+  if (moment(d, 'YYYY-MM-DD HH:mm:ss', true).isValid()) {
+    return 'YYYY-MM-DD HH:mm:ss'
+  }
+  if (moment(d, 'YYYY-MM-DD HH:mm:ss.SSS', true).isValid()) {
+    return 'YYYY-MM-DD HH:mm:ss.SSS'
+  }
+  return
+}
+
+export class DatePicker extends PureComponent<Props, State> {
+  private inCurrentMonth: boolean = false
+
+  public static defaultProps = {
+    testID: 'date-picker',
+  }
+
+  state = {
+    inputValue: '',
+    inputFormat: '',
+  }
+
+  public render() {
+    const {dateTime, label, testID} = this.props
+
+    const date = new Date(dateTime || Date.now())
+
+    return (
+      <div className="date-picker" data-testid={testID}>
+        <Grid.Row>
```

Comment:
Does this component need `Grid` to work?

Reply:
Nope, removing